mbpfan.spec

```diff
@@ -0,0 +1,68 @@
+Name:           mbpfan
+URL:            https://github.com/dgraziotin/mbpfan
+License:        GPLv3
+Group:          System Environment/Daemons
+Version:        %{SOURCE_VERSION}
+Release:        1
+Summary:        A simple daemon to control fan speed on all MacBook/MacBook Pros (probably all Apple computers) for Linux 3.x.x and 4.x.x
+Source:         v%{version}.tar.gz
+
+%description
+This is an enhanced version of Allan McRae mbpfan
+
+mbpfan is a daemon that uses input from coretemp module and sets the fan speed using the applesmc module. This enhanced version assumes any number of processors and fans (max. 10).
+
+It only uses the temperatures from the processors as input.
+It requires coretemp and applesmc kernel modules to be loaded.
+It requires root use
+It daemonizes or stays in foreground
+Verbose mode for both syslog and stdout
+Users can configure it using the file /etc/mbpfan.conf
+
+%prep
+%setup -q -n %{name}-%{version}
+
+%build
+make
+
+%install
+install -D -m755 bin/mbpfan $RPM_BUILD_ROOT/usr/sbin/mbpfan
+install -D -m644 mbpfan.conf $RPM_BUILD_ROOT/etc/mbpfan.conf
+install -D -m644 mbpfan.service $RPM_BUILD_ROOT/usr/lib/systemd/system/mbpfan.service
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%post
+%systemd_post mbpfan.service
+# If it is a first installation then autoconfigure daemon
+if [[ ${1} == 1 ]]; then
+  min_fan_speed="$(cat /sys/devices/platform/applesmc.768/fan*_min| uniq| sort |head -1)"
+  max_fan_speed="$(cat /sys/devices/platform/applesmc.768/fan*_max| uniq| sort |tail -1)"
+  let max_temp=$(cat /sys/devices/platform/coretemp.*/hwmon/hwmon*/temp*_max|uniq |sort |tail -1)/1000
+  sed -i "s/min_fan_speed = [^\t]\+/min_fan_speed = $min_fan_speed/g" /etc/mbpfan.conf
+  sed -i "s/max_fan_speed = [^\t]\+/max_fan_speed = $max_fan_speed/g" /etc/mbpfan.conf
+  sed -i "s/max_temp = [^\t]\+/max_temp = $max_temp/g" /etc/mbpfan.conf
+fi
+echo "Attention: mbpfan preconfigured with sane values, to run it type:"
+echo "systemctl start mbpfan"
+echo "To run also at boot, type:"
+echo "systemctl enable mbpfan"
```

Comment:
I'm trying to understand the Fedora recommendation in #154.  Do services like postgres require this manual systemd intervention?

Reply:
here is some details about it: https://www.freedesktop.org/software/systemd/man/systemd.preset.html
>>Preset files may be used to encode policy which units shall be enabled by default and which ones shall be disabled.
so, if you don't create special file, which will define service as enabled it will disabled by default and used need to enable it (so make in runnable) and start(if it's required)

Reply:
Every service/daemon that need to be configures by user (or root) should not be started automatically. The reason is this: If I start a service (like postgres as you suggested) right after the installation without reconfiguring it might have some undesirable settings or dangerous one (il was of postgres it will run with the default administrative password). Il our case the mbpfan service will be preconfigured with the recommended settings but the user need to review them. Maybe I should re-phrase the warning at the end of the posinstall scriptlet.
Let me know.

Reply:
I will defer to you both -- I don't understand systemd well enough to have an opinion.